src/lib.rs

```diff
@@ -0,0 +1,408 @@
+#![forbid(unsafe_code)]
+
+mod baseline;
+mod check_release;
+mod config;
+mod dump;
+mod manifest;
+mod query;
+mod templating;
+mod util;
+
+use cargo_metadata::PackageId;
+pub use config::*;
+pub use query::*;
+
+use check_release::run_check_release;
+use trustfall_rustdoc::{load_rustdoc, VersionedCrate};
+
+use dump::RustDocCommand;
+use itertools::Itertools;
+use semver::Version;
+use std::collections::HashSet;
+use std::path::{Path, PathBuf};
+
+/// Test a release for semver violations.
+pub struct Check {
```

Comment:
We should probably `#[derive(Debug)]` and `#[non_exhaustive]` at least for all these pub structs and enums, unless there's a reason not to?

Reply:
Do you want to add `#[non_exhaustive]` even if the struct fields are private?
I don't think it has any effect in this case.

Reply:
done! :)